code/modules/admin/abilitymanager.dm

```diff
@@ -0,0 +1,80 @@
+/datum/abilitymanager
+	var/mob/target_mob
+
+/datum/abilitymanager/ui_state(mob/user)
+	return tgui_admin_state.can_use_topic(src, user)
+
+/datum/abilitymanager/ui_status(mob/user)
+	return tgui_admin_state.can_use_topic(src, user)
+
+/datum/abilitymanager/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "AbilityManager")
+	ui.open()
+
+/datum/abilitymanager/ui_data(mob/user)
+	var/list/ability_props
```

Comment:
You probably want to initialize this empty?

Reply:
The JS is able to handle receiving it as either null or empty, I can initialize it as empty to be clearer though